src/core/dense_set.h

```diff
@@ -187,6 +187,14 @@ class DenseSet {
       return curr_entry_->HasTtl() ? owner_->ObjExpireTime(curr_entry_->GetObject()) : UINT32_MAX;
     }
 
+    void SetExpiryTime(uint32_t ttl_sec) {
+      if (HasExpiry()) {
+        owner_->ObjUpdateExpireTime(curr_entry_->GetObject(), ttl_sec);
+      } else {
+        owner_->ObjReplace(curr_entry_->GetObject(), ttl_sec);
```

Comment:
`owner_->AddOrUpdate()` and you can delete the function `ObjReplace()`

Reply:
If I'm to do `owner_->AddOrUpdate()` I would have to expose that as a new virtual function on `dense_set` though?
Which I can obviously do, as it would remove a code-path on `string_map`, but it just seems like your comment assumes `AddOrUpdate` already exists, which confuses me a bit.

Reply:
Sorry, I was not very clear with this -- I wanted `SetExpiryTime()` to be a member of the derived class and not a member of the iterator `DenseSet::Iterator::`. Based on Roman's comments, my suggestion is to use `AddOrUpdate()` here + make it virtual since Roman does not want to implement this in the derived class.

He also suggested a different approach below (see his comment below). I replied there. Let him decide out of the two approaches which one he prefers and we move on. Both of them do not require a lot of changes and they should be pretty quick to implement. 

Also @NegatioN again thank you for your contribution and for your patience with this :pray: :rocket: